packages/media-utils/src/components/media-upload/index.js

```diff
@@ -396,10 +396,11 @@ class MediaUpload extends Component {
 
 	onClose() {
 		const { onClose } = this.props;
-
 		if ( onClose ) {
 			onClose();
 		}
+
+		this.isModalOpen = false;
```

Comment:
I'm no Media Library code expert, but I think we should call `this.frame.detach()` on close action. Otherwise, the modal leaves some "breadcrumbs" in the DOM. Core does a similar thing for [classic widgets](https://github.com/WordPress/wordpress-develop/blob/trunk/src/js/_enqueues/wp/widgets/media-audio.js#L142-L144).

## Screenshot
![CleanShot 2024-05-29 at 19 37 13](https://github.com/WordPress/gutenberg/assets/240569/f7214f38-5bd1-4509-8af9-208c4ab517e9)

Reply:
Left a longer reply [in Slack](https://wordpress.slack.com/archives/C02SX62S6/p1717096254809339?thread_ts=1717055547.327059&cid=C02SX62S6). TL;DR I agree.